server/handlers/handlers.go

```diff
@@ -101,6 +102,36 @@ func (ah *ActivityHandler) Start() {
 		}
 	}
 
+	// doStart is the meat of the activity handler worker
+	doStartLastActivityAt := func() {
+		for {
+			timer := time.NewTimer(5 * time.Minute)
+			select {
+			case <-timer.C:
+				ah.lastUpdateAtMap.Range(func(subscriptionID, lastUpdateAt any) bool {
+					if lastUpdateAt.(time.Time).Add(5 * time.Minute).After(time.Now()) {
+						if err := ah.plugin.GetStore().UpdateSubscriptionLastActivityAt(subscriptionID.(string), lastUpdateAt.(time.Time)); err != nil {
+							ah.plugin.GetAPI().LogWarn("Error storing the subscription last activity at", "error", err)
+						}
+					}
+					return true
+				})
+			case <-ah.quit:
+				timer.Stop()
+				// we have received a signal to stop
+				return
```

Comment:
Should we flush our cache to the database when the worker exits?

Reply:
We could.